### PR TITLE
Add other Meridio resources to be managed by operator

### PR DESCRIPTION
**Pull request by mandydydy**
Meridio should be initialized by applying a custom resource
All relevant resources should be created and updated by the custom resource.

- [x] a framework for all resources
- [x] ipam deployment and service
- [x] configmap
- [x] load-balancer deployment
- [x] nse-vlan
- [x] nsp-service and deployment
- [x] proxy

### CODE REVIEW

File: config/manager/manager.yaml [hunk 1]
```diff
@@ -30,6 +30,7 @@ spec:
         args:
         - --leader-elect
         image: controller:latest
+        imagePullPolicy: Never
```
Comment (LionelJouin) [inline config/manager/manager.yaml:33]: I had some troubles to deploy the operator, maybe "IfNotPresent" would be preferable?

**Review by LionelJouin — CHANGES_REQUESTED** [review]

File: controllers/trench_controller.go [hunk 2]
```diff
@@ -68,109 +66,28 @@ func (r *TrenchReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
...
-}
-
 // SetupWithManager sets up the controller with the Manager.
 func (r *TrenchReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&meridiov1alpha1.Trench{}).
 		Owns(&corev1.ConfigMap{}).
+		Owns(&corev1.Service{}).
+		Owns(&appsv1.Deployment{}).
 		Complete(r)
```
Comment (LionelJouin) [inline controllers/trench_controller.go:92]: More resources are deployed by the operator (Role, Rolebinding, Service account...), should they be added here?

File: deployment/ipam-service.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: ipam
```
Comment (LionelJouin) [inline deployment/ipam-service.yaml:4]: Should be "ipam-service" instead of "ipam"

**Review by LionelJouin — CHANGES_REQUESTED** [review]

File: controllers/trench_controller.go (hunk 2 above)
Comment (mandydydy) [reply controllers/trench_controller.go:97]: All resouces should be added now

File: config/manager/manager.yaml (hunk 1 above)
Comment (mandydydy) [reply config/manager/manager.yaml]: This is reverted now

**Review by LionelJouin — APPROVED** [review]

**Pull request merged**